compiler/src/iree/compiler/DispatchCreation/SetEncoding.cpp

```diff
@@ -276,12 +309,16 @@ struct FoldFillWithSetEncoding final
   }
 };
 
-struct SetEncodingPass final : impl::SetEncodingPassBase<SetEncodingPass> {
-  using Base::Base;
+class SetEncodingPass : public impl::SetEncodingPassBase<SetEncodingPass> {
+public:
+  using impl::SetEncodingPassBase<SetEncodingPass>::SetEncodingPassBase;
+  explicit SetEncodingPass(const EncodingOptions &option) {
+    this->encodingOption = option;
+  }
```

Comment:
I think we don't need this, right? The `using Base::Base` should already provide the constructor.